apis/kueue/v1beta1/admissioncheck_types.go

```diff
@@ -25,16 +25,14 @@ type CheckState string
 const (
 	// CheckStateRetry means that the check cannot pass at this moment, back off (possibly
 	// allowing other to try, unblock quota) and retry.
-	// A workload having at least one check in the state,
-	// will be evicted if admitted will not be considered
-	// for admission.
+	// A workload having at least one check in the state, will be evicted if admitted and
+	// not be considered for admission.
```

Comment:
```suggestion
	// will no longer be considered for admission.
```

Reply:
or is it:

```suggestion
	// will not be considered for admission while the check is in this state.
```
?

Please refresh my memory: how does Kueue get out of this state?

Reply:
The AC controller should switch it to any other state by its own logic.

changed to `will not be considered for admission.`

Reply:
Indeed the current status is that when `Retry` is set on a workload's admission check, then it stays, so the workload is not considered for admission, until flipped to `Pending`. 

However, I did some digging into the KEP (as this was confusing to me during review of https://github.com/kubernetes-sigs/kueue/pull/1681 which starts to make use of `Retry`), and at the API level we have `RetryDelayMinutes`, IIUC this should automatically transition the state from `Retry` to `Pending`, so that the workload can be re-admitted after 15min. I think it is currently not supported though. 

Let me know if I'm not missing something. I think it would be good to support `RetryDelayMinutes` as it is part of the API (surely separate PR). If we have other priorities I would suggest mentioning that this is the current state for Alpha.

Reply:
I do nor recall the exact circumstances `RetryDelayMinutes` did not make it in the implementation, the preemtion support is not yet merged also being still in review. 

Going back to `RetryDelayMinutes` that approach is very inflexible,  different workloads may need different times to wait before requeue. 

Anyway this discussion is not relevant in the context of this PR.

Reply:
I find it relevant because if we supported the retry delay field, then the workload is considered for admission after that time. I think the comments for the retry field and the fields you amend in this comment are not compatible currently.

If we don't aim to support retry delay field in near future then I would suggest to update its comment so that the comments are aligned.

Reply:
done